pkg/v1/v1util/zip.go

```diff
@@ -80,6 +84,27 @@ func GunzipReadCloser(r io.ReadCloser) (io.ReadCloser, error) {
 	}, nil
 }
 
+// EstargzReadCloser reads uncompressed input data from the io.ReadCloser and
+// returns an io.ReadCloser from which compressed data may be read.
+// Refer to estargz for the options:
+// https://pkg.go.dev/github.com/containerd/stargz-snapshotter@v0.2.0/estargz#Option
+func EstargzReadCloser(r io.ReadCloser, opts ...estargz.Option) (io.ReadCloser, v1.Hash, error) {
+	defer r.Close()
+
+	bs, err := ioutil.ReadAll(r)
```

Comment:
Assuming this is just for prototyping and we won't always buffer the whole thing for a final implementation? We should be teeing `r` into a hasher, right?

Reply:
Yeah, I don't love this implementation either, but I think the estargz stuff wants to be able to seek around to build the index and reorder things for the final estargz file.  The `v1.Hash` is not the overall hash, but the hash of the table of contents.  I owe a better function comment than this has, so let me take a crack at that now.